File: tests/scala/src/main/scala/struct/versions/Main_Conversions.scala
package struct.versions

import dev.argon.verilization.scala_runtime.Converter

object Main_Conversions {
    val v3ToV4: Converter[Main.V3, Main.V4] = prev =>
        Main.V4(
            n = prev.n,
            m = prev.m,
            r = Referenced.V4.fromV3.convert(prev.r),
            addition = Addition.V4(5),
        )
}

File: tests/java/src/main/java/struct/versions/Main_Conversions.java
package struct.versions;

import dev.argon.verilization.java_runtime.Converter;

final class Main_Conversions {
    private Main_Conversions() {}
    
    static final Converter<Main.V3, Main.V4> v3ToV4 = new Converter<Main.V3, Main.V4>() {
        @Override
        public Main.V4 convert(Main.V3 prev) {
            return new Main.V4(
                prev.n,
                prev.m,
                Referenced.V4.fromV3.convert(prev.r),
                new Addition.V4(
                    5
                )
            );
        }
    };

}
